src/sentry/search/snuba/backend.py

```diff
@@ -1,72 +1,52 @@
+from sentry.api.event_search import InvalidSearchQuery
+from sentry.models import Release, GroupEnvironment, Group, GroupStatus, GroupSubscription
 from sentry.search.base import SearchBackend
-from sentry.utils import snuba, metrics
-
-logger = logging.getLogger("sentry.search.snuba")
-datetime_format = "%Y-%m-%dT%H:%M:%S+00:00"
-
-EMPTY_RESULT = Paginator(Group.objects.none()).get_result()
-
-# mapping from query parameter sort name to underlying scoring aggregation name
-sort_strategies = {
-    "date": "last_seen",
-    "freq": "times_seen",
-    "new": "first_seen",
-    "priority": "priority",
-}
-
-dependency_aggregations = {"priority": ["last_seen", "times_seen"]}
-
-aggregation_defs = {
-    "times_seen": ["count()", ""],
-    "first_seen": ["multiply(toUInt64(min(timestamp)), 1000)", ""],
-    "last_seen": ["multiply(toUInt64(max(timestamp)), 1000)", ""],
-    # https://github.com/getsentry/sentry/blob/804c85100d0003cfdda91701911f21ed5f66f67c/src/sentry/event_manager.py#L241-L271
-    "priority": ["toUInt64(plus(multiply(log(times_seen), 600), last_seen))", ""],
-    # Only makes sense with WITH TOTALS, returns 1 for an individual group.
-    "total": ["uniq", "issue"],
-}
-issue_only_fields = set(
-    [
-        "query",
-        "status",
-        "bookmarked_by",
-        "assigned_to",
-        "unassigned",
-        "subscribed_by",
-        "active_at",
-        "first_release",
-        "first_seen",
-    ]
-)
+from sentry.search.snuba.executors import PostgresSnubaQueryExecutor, SnubaOnlyQueryExecutor
 
 
-class QuerySetBuilder(object):
-    def __init__(self, conditions):
-        self.conditions = conditions
+def assigned_to_filter(actor, projects):
+    from sentry.models import OrganizationMember, OrganizationMemberTeam, Team
 
-    def build(self, queryset, search_filters):
-        for search_filter in search_filters:
-            name = search_filter.key.name
-            if name in self.conditions:
-                condition = self.conditions[name]
-                queryset = condition.apply(queryset, search_filter)
-        return queryset
+    if isinstance(actor, Team):
+        return Q(assignee_set__team=actor)
+
+    teams = Team.objects.filter(
+        id__in=OrganizationMemberTeam.objects.filter(
+            organizationmember__in=OrganizationMember.objects.filter(
+                user=actor, organization_id=projects[0].organization_id
+            ),
+            is_active=True,
+        ).values("team")
+    )
+
+    return Q(
+        Q(assignee_set__user=actor, assignee_set__project__in=projects)
+        | Q(assignee_set__team__in=teams)
+    )
+
+
+def unassigned_filter(unassigned, projects):
+    from sentry.models.groupassignee import GroupAssignee
+
+    query = Q(
+        id__in=GroupAssignee.objects.filter(project_id__in=[p.id for p in projects]).values_list(
+            "group_id", flat=True
+        )
+    )
+    if unassigned:
+        query = ~query
+    return query
```

Comment:
I think this is already in master. Could you please merge it.

Reply:
Unresolving this because I don;t think the comment was addressed. 
This PR is not based on the current master. It seems it is missing major changes to snuba/backend.py that are included in PR https://github.com/getsentry/sentry/pull/15708/files that was already merged.

Please either rebase or merge master into this branch, till then, reviewing is not very useful.